src/offset_of.rs

```diff
@@ -49,7 +49,7 @@ macro_rules! _memoffset__let_base_ptr {
 #[macro_export]
 #[doc(hidden)]
 macro_rules! _memoffset__field_check {
-    ($type:tt, $field:tt) => {
+    ($type:path, $field:tt) => {
```

Comment:
Last time I tried this this didn't work because of pattern stuff... but CI seems happy? Do you now which issue that could have been?

Reply:
I have always used `path` in `intrusive-collections` and never had any issues.

Reply:
Well, CI is happy here as well, so LGTM. Would have preferred this as a separate PR, though.